src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -151,26 +152,36 @@ public IWebHost Build()
             }
             _webHostBuilt = true;
 
-            // Warn about deprecated environment variables
-            if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
-            {
-                Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
-            }
+            var hostingServices = BuildCommonServices(out var hostingStartupErrors);
+            var applicationServices = hostingServices.Clone();
+            var hostingServiceProvider = hostingServices.BuildServiceProvider();
 
-            if (Environment.GetEnvironmentVariable("ASPNET_ENV") != null)
+            if (_options.WriteStatusMessages)
             {
-                Console.WriteLine("The environment variable 'ASPNET_ENV' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
+                // Warn about deprecated environment variables
+                if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
+                {
+                    Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
+                }
+
+                if (Environment.GetEnvironmentVariable("ASPNET_ENV") != null)
+                {
+                    Console.WriteLine("The environment variable 'ASPNET_ENV' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
+                }
+
+                if (Environment.GetEnvironmentVariable("ASPNETCORE_SERVER.URLS") != null)
+                {
+                    Console.WriteLine("The environment variable 'ASPNETCORE_SERVER.URLS' is obsolete and has been replaced with 'ASPNETCORE_URLS'");
+                }
             }
 
-            if (Environment.GetEnvironmentVariable("ASPNETCORE_SERVER.URLS") != null)
+            var logger = hostingServiceProvider.GetRequiredService<ILogger<WebHost>>();
```

Comment:
For my own knowledge, should we be getting the required logger here? I thought the whole point of Console.Writeline was to avoid pulling in the logger.

Reply:
Nvrm, read feedback above.